### ISSUE TITLE
Not working properly on android/ios

### ISSUE THREAD

**Issue by SupertigerDev** [preactjs/signals#571]
- [ ] Check if updating to the latest version resolves the issue

**Environment**
- [ ] I am using `@preact/signals-core`
- [ ] I am using `@preact/signals`
- [x] I am using `@preact/signals-react`
  - [x] I understand usage changed in v2, and I've followed the [React Integration instructions](https://github.com/preactjs/signals/tree/chore/issue-templates/packages/react#react-integration)

**Describe the bug**
On web, it works fine. the input field shows the existing signal value. but on ios/android, the initial signal value does not show.

**To Reproduce**
https://snack.expo.dev/PWLDLLtxZpvpWq2hW_OKT?platform=web

1. Go to the url above
2. go to the web version and see that its working as intended, the input field contains "test"
3. go to android/ios, see that the input field is empty. 

**Expected behavior**
The input field for ios/android should contain the word "test" just like in the web version.

**Comment by rschristian** [comment]
Looks like unwrapping the signal works, issue with the binding in RN I guess.

**Comment by SupertigerDev** [comment]
> Looks like unwrapping the signal works, issue with the binding in RN I guess.

Unwrapping it also has bugs. I can't input text on android/iPhone

**Comment by rschristian** [comment]
You checked the "I understand usage changed in v2, and I've followed the [React Integration instructions](https://github.com/preactjs/signals/tree/main/packages/react#react-integration)" box yet I don't see any indication of using the Babel plugin or `useSignals()`

**Comment by SupertigerDev** [comment]
Oh my apologies, I assumed the stuff in readme was the latest. I will try again with useSignal and give you an update later.

**Comment by SupertigerDev** [comment]
Is the Babel stuff optional if I just use useSignal?

**Comment by rschristian** [comment]
The ReadMe does document the latest usage, you just seem to have skipped over it.

`useSignals()`, plural. You need to use it or the Babel plugin.

**Comment by SupertigerDev** [comment]
Oh, by the way, the hyperlink for the issue template `React Integration instructions` points to a 404 page

**Comment by rschristian** [comment]
Yup, realized that. My bad #573

**Comment by SupertigerDev** [comment]
After looking at the initial expo link again, it seems like i am using `useSignal`. I'm kind of confused what you mean 🤔 
![image](https://github.com/preactjs/signals/assets/8193841/adab5d45-7339-4a89-828a-b03afa29146e)

**Comment by SupertigerDev** [comment]
Ah, seems like I missread something. there is `useSignal` and `useSignals` which i did not realize because of their names being VERY similar. My bad.

Although, I'm not sure what is going on here. Is this an expo issue?
![image](https://github.com/preactjs/signals/assets/8193841/ce1f7b3e-d230-4375-be27-bffc3433e543)

**Comment by rschristian** [comment]
Looks like they don't support package exports? That's a bit rough, I suppose you can try forming [the path](https://github.com/preactjs/signals/blob/609a2b38947ac2d0fed49175813f0033a4a26586/packages/react/package.json#L37) manually?

**Comment by SupertigerDev** [comment]
Annoyingly that didnt work either 😦 let me really quickly create a new react native project without expo 1 sec

**Comment by SupertigerDev** [comment]
![explorer_aNbSnmnWZb](https://github.com/preactjs/signals/assets/8193841/610d728d-a54f-4c2e-8220-117a629fabdd)
Well, its working, however its not finegrained. Other elements also update. Unless its not intended to be finegrain like solidjs?
```ts
export default function App() {
  useSignals();
  return (
    <SafeAreaView style={styles.container}>
      <TextInput
        style={styles.input}
        value={testSignal.value}
        onChangeText={e => (testSignal.value = e)}
      />
      <Text>
        <>{testSignal}</>
      </Text>
      <Text>{Math.random()}</Text>
    </SafeAreaView>
  );
}
```

**Comment by rschristian** [comment]
> however its not finegrained

Yes, if you unwrap with `.value` you de-opt into full component renders.

However, if RN is borked then this is all that I can recommend at the moment. RN isn't something we directly support / test against.

**Comment by SupertigerDev** [comment]
yea its broken when i dont unwrap 😭
